src/prefect/tasks/aws/step_function.py

```diff
@@ -12,16 +15,16 @@ class StepActivate(Task):
         - execution_name (str): the name of the execution, this name must be unique for
             your AWS account, region, and state machine for 90 days
```

Comment:
Would it be appropriate to add a deprecation or warning to this argument? I can imagine others falling into the same trap, where the task runs fine locally, but fails on the second execution in production because `execution_name` is set at registration time.

Reply:
Yes, I think it's a great idea to note in the docstring that `execution_name` should be set when running the task instead of when instantiating a task instance and also log a warning if `execution_name` is provided during instantiation. @madkinsz can we remove `execution_name` from the `__init__` method in the 1.0 release?

Reply:
Yeah we can remove it then